File: Лабы_1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int spaces, int bricks);

int main(void)
{
    // Указание высоты пирамиды пользователем
    int height;
    do
    {
        height = get_int("Введите высоту пирамиды: ");
    }
    while (height < 1 || height > 8);

    // Вывод пирамиды
    for (int i = 0; i < height; i++)
    {
        // Вывод строк с кирпичами
        print_row(height - i, i + 1);
    }
}

void print_row(int spaces, int bricks)
{
    // Вывод пробелов
    for (int i = 0; i < spaces - 1; i++)
    {
        printf(" ");
    }

    for (int j = 0; j < bricks; j++)
    {
        printf("#");
    }

    printf("\n");
}

File: Лабы_6/sentimental-mario-more/mario.py
from cs50 import get_int

def main():
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break

    for i in range(height):
        print_row(height - i, i + 1)

# Печать всех сторон пирамиды, начиная с левой
def print_row(spaces, bricks):
    for i in range(spaces - 1):
        print(" ", end="")

    for j in range(bricks):
        print("#", end="")

    print(" " * 2, end="")

    print_row_right(bricks)

# Печать правой стороны пирамиды
def print_row_right(bricks):
    for i in range(bricks):
        print("#", end="")

    print()

main()
